toolkits/kaldi/gigaspeech_data_prep.sh

```diff
@@ -4,32 +4,111 @@
+      fi
+      eval $name=\"$2\";
+      if $was_bool && [[ "$2" != "true" && "$2" != "false" ]]; then
+        echo "$0: expected \"true\" or \"false\": $1 $2" 1>&2
+        exit 1;
+      fi
+      shift 2;
+      ;;
+  *) break;
+  esac
+done
+
+filter_by_id () {
+  idlist=$1
+  input=$2
+  output=$3
+  field=1
+  if [ $# -eq 4 ]; then
+    field=$4
+  fi
+  cat $input | perl -se '
+    open(F, "<$idlist") || die "Could not open id-list file $idlist";
+    while(<F>) {
+      @A = split;
+      @A>=1 || die "Invalid id-list file line $_";
+      $seen{$A[0]} = 1;
+    }
+    while(<>) {
+      @A = split;
+      @A > 0 || die "Invalid file line $_";
+      @A >= $field || die "Invalid file line $_";
+      if ($seen{$A[$field-1]}) {
+        print $_;
+      }
+    }' -- -idlist="$idlist" -field="$field" > $output ||\
+  (echo "$0: Failed to call filter_by_id(): $input" && exit 1)
+}
 
-if [ $# -lt 3 ] || [ $# -gt 4 ]; then
-  echo "Usage: $0 <gigaspeech-dataset-dir> <data-dir> <use-pipe> [<subset-prefix>]"
-  echo " e.g.: $0 /disk1/audio_data/gigaspeech data/ true gigaspeech"
+subset_data_dir () {
+  utt_list=$1
+  src_dir=$2
+  dest_dir=$3
+  mkdir -p $dest_dir || exit 1
+  # wav.scp utt2spk text segments utt2dur reco2dur spk2utt
+  filter_by_id $utt_list $src_dir/utt2spk $dest_dir/utt2spk ||\
+    (echo "$0: Failed to call subset_data_dir(): $src_dir/utt2spk" && exit 1)
+  filter_by_id $utt_list $src_dir/spk2utt $dest_dir/spk2utt 2 ||\
+    (echo "$0: Failed to call subset_data_dir(): $src_dir/spk2utt" && exit 1)
+  filter_by_id $utt_list $src_dir/utt2dur $dest_dir/utt2dur ||\
+    (echo "$0: Failed to call subset_data_dir(): $src_dir/utt2dur" && exit 1)
+  filter_by_id $utt_list $src_dir/text $dest_dir/text ||\
+    (echo "$0: Failed to call subset_data_dir(): $src_dir/text" && exit 1)
+  filter_by_id $utt_list $src_dir/segments $dest_dir/segments ||\
+    (echo "$0: Failed to call subset_data_dir(): $src_dir/segments" && exit 1)
+  awk '{print $2}' $dest_dir/segments | sort | uniq > $dest_dir/reco
+  filter_by_id $dest_dir/reco $src_dir/wav.scp $dest_dir/wav.scp ||\
+    (echo "$0: Failed to call subset_data_dir(): $src_dir/wav.scp" && exit 1)
+  filter_by_id $dest_dir/reco $src_dir/reco2dur $dest_dir/reco2dur ||\
+    (echo "$0: Failed to call subset_data_dir(): $src_dir/reco2dur" && exit 1)
+  rm -f $dest_dir/reco
+}
+
+if [ $# -ne 2 ]; then
+  echo "Usage: $0 [options] <gigaspeech-dataset-dir> <data-dir>"
+  echo " e.g.: $0 --train-subset XL /disk1/audio_data/gigaspeech/ data/"
   echo ""
   echo "This script takes the GigaSpeech source directory, and prepares the"
-  echo "Kaldi format data directory. When <use-pipe> is true, it decodes the"
-  echo "OPUS audio format through a pipe; Otherwise it writes the decoded"
-  echo "output to wav files. <subset-prefix> is an optional prefix for Kaldi"
-  echo "data directories."
+  echo "Kaldi format data directory."
+  echo "  --garbage-tags <tags>        # Tags for non-speech."
```

Comment:
I think renaming this option to "garbage-utterance-tags" is more clear to users, otherwise people might think these tags represent "words" similar to "punctuation-tags"

Reply:
Done.